core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -436,14 +436,14 @@ private[spark] class Executor(
    * This thread stops running when the executor is stopped.
    */
   private def startDriverHeartbeater(): Unit = {
-    val interval = conf.getInt("spark.executor.heartbeatInterval", 10000)
+    val intervalMs = Utils.timeStringAsMs(conf.get("spark.executor.heartbeatInterval", "10s"))
     val thread = new Thread() {
       override def run() {
-        // Sleep a random interval so the heartbeats don't end up in sync
-        Thread.sleep(interval + (math.random * interval).asInstanceOf[Int])
+        // Sleep a random intervalMs so the heartbeats don't end up in sync
```

Comment:
I don't think the "interval" here was referring to the variable, so could you please revert this :)